### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
Ran `make update-dependencies`.

#### Ticket Link
--

### CODE REVIEW

**Comment by agarciamontoro** [comment]
PR #700, let's celebrate! :tada:

**Review by agnivade — APPROVED** [review]

**Comment by agarciamontoro** [comment]
For some reason, using go 1.19 made golangci-lint fail with this error:
```
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package model: could not load export data: no export data for "github.com/prometheus/common/model"
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: buildir: failed to load package model: could not load export data: no export data for "github.com/prometheus/common/model"
```

I didn't catch this locally because my local version was 1.20. I'm not sure why 1.19 triggered this error, but an update has been pending for a while in any case, so here it is.

Re-requesting review for the new changes.

**Review by agnivade — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
         with:
-          go-version: "1.19.5"
+          go-version: "1.20.14"
```
Comment (streamer45) [inline .github/workflows/ci.yml:23]: Is the dot release actually needed here?

**Review by streamer45 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (agarciamontoro) [reply .github/workflows/ci.yml:23]: Not sure, to be honest. It was a quick fix so I just copied the same pattern. Let me check.

Comment (agarciamontoro) [reply .github/workflows/ci.yml]: It seems it's not needed: https://github.com/marketplace/actions/setup-go-environment#supported-version-syntax Let's see in practice

Comment (agarciamontoro) [reply .github/workflows/ci.yml]: It works :+1:

**Pull request merged**